### PR TITLE
Text exporter encounters

### PR DESCRIPTION
**Pull request by laambrosino**
The other branch was messy in general so I deleted it and created this one...  I think there are still conflicts so I can try rebasing like you suggested

### CODE REVIEW

File: src/main/resources/synthea.properties [hunk 1]
```diff
@@ -14,9 +14,12 @@ exporter.fhir.transaction_bundle = false
 exporter.hospital.fhir.export = true
 exporter.hospital.fhir_dstu2.export = false
 exporter.csv.export = false
+
 exporter.text.export = false
+exporter.text.export2 = false
```
Comment (dehall) [inline src/main/resources/synthea.properties]: Could you use a more descriptive name than export2? It's not immediately clear what this means. Maybe `exporter.text.per_encounter_export` or something like that.

File: src/main/java/org/mitre/synthea/export/TextExporter.java [hunk 2]
```diff
@@ -242,8 +401,33 @@ private static void basicInfo(List<String> textRecord, Person person, long endTi
...
+   * @param person
+   *          The person to export
+   * @param encounter
+   *          The encounter all of the information refers to
+   */
+  private static void encounterReport(List<String> textRecord, Person person, Encounter encounter) {
+    String encounterTime = dateFromTimestamp(encounter.start);
 
-    if (encounter.reason == null && encounter.provider == null) {
+    if (encounter.reason == null){
```
Comment (dehall) [inline src/main/java/org/mitre/synthea/export/TextExporter.java]: Looks like this `&& encounter.provider == null)` got lost in the rebase. This should still be in there.

File: src/main/java/org/mitre/synthea/export/TextExporter.java [hunk 3]
```diff
@@ -263,18 +532,44 @@ private static void encounter(List<String> textRecord, Encounter encounter) {
...
+   */
+  private static void conditionpast(List<String> textRecord, Entry condition, Encounter encounter) {
+    String start = dateFromTimestamp(condition.start);
+    if ((condition.stop == 0L || condition.stop > encounter.stop) && (condition.start < encounter.start)) {
+      //checks that the condition hasn't ended by the time of the encounter
+      //and began prior to the encounter
+      String description = condition.codes.get(0).display;
+        textRecord.add("  " + start + " : " + description);
+    } else {
+      return;
```
Comment (dehall) [inline src/main/java/org/mitre/synthea/export/TextExporter.java]: Don't need to explicitly return in a case like this - you can just omit the `else`

File: src/main/java/org/mitre/synthea/export/TextExporter.java [hunk 4]
```diff
@@ -188,7 +187,167 @@ public static void export(Person person, long time) throws IOException {
...
+   *                        Activity: Inhaled steroid therapy
+   *                        Activity: Home nebulizer therapy
+   *                        Activity: Breathing control
+   *   
+   * --------------------------------------------------------------------------------
+   */
+  
+  public static void exportEncounter(Person person, long time) throws IOException {
+  
+  /**
```
Comment (dehall) [inline src/main/java/org/mitre/synthea/export/TextExporter.java]: This javadoc comment should be above the method signature

File: src/main/resources/synthea.properties [hunk 5]
```diff
@@ -14,9 +14,12 @@ exporter.fhir.transaction_bundle = false
 exporter.hospital.fhir.export = true
 exporter.hospital.fhir_dstu2.export = false
 exporter.csv.export = false
-exporter.text.export = false
+
+exporter.text.export = true
```
Comment (jawalonoski) [inline src/main/resources/synthea.properties]: Text exporters should default to `false`

File: src/main/java/org/mitre/synthea/export/Exporter.java [hunk 6]
```diff
@@ -287,4 +295,13 @@ public static String filename(Person person, String extension) {
           + extension;
     }
   }
+
+  public static String filename2(Person person, String encounterNumber, String extension) {
```
Comment (jawalonoski) [inline src/main/java/org/mitre/synthea/export/Exporter.java]: `filename2` is not a good name (try not to ever use numbers in a function name) -- try something more descriptive like `filename_per_encounter(...)` or something.

File: src/main/java/org/mitre/synthea/export/TextExporter.java [hunk 7]
```diff
@@ -263,18 +518,42 @@ private static void encounter(List<String> textRecord, Encounter encounter) {
    * @param condition
    *          The condition to add to the export
    */
-  private static void condition(List<String> textRecord, Entry condition) {
+  private static void condition(List<String> textRecord, Entry condition, Boolean end) {
```
Comment (dehall) [inline src/main/java/org/mitre/synthea/export/TextExporter.java]: One last request before we merge this - for these methods where you've added a new `end` parameter please add a description of the parameter to the javadoc above the method.

**Pull request merged**